src/test/scala/uk/gov/nationalarchives/tdr/ErrorHandlingTest.scala

```diff
@@ -0,0 +1,136 @@
+package uk.gov.nationalarchives.tdr
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.matchers.should.Matchers
+import sttp.model.StatusCode
+import uk.gov.nationalarchives.tdr.GraphQLClient.Location
+import uk.gov.nationalarchives.tdr.error.{GraphQlClientException, HttpException, NotAuthorisedError, UnknownGraphQlError}
+import uk.gov.nationalarchives.tdr.testdata.AddFileTestDocument.addFile.addFileDocument
+import uk.gov.nationalarchives.tdr.testdata.GetSeriesTestDocument.getSeries.{GetSeriesVariables, SeriesResponseData}
+
+import scala.io.Source.fromResource
+
+class ErrorHandlingTest extends WireMockTest with Matchers {
+
+  def getSeriesClient = new GraphQLClient[SeriesResponseData, GetSeriesVariables](graphQlUrl)
+
+  "a 401 Unauthorized response" should "cause an HTTP error" in {
+    wiremockServer.stubFor(post(urlEqualTo(graphQlPath))
+      .willReturn(unauthorized().withBody("some response body")))
+
+    val result = getSeriesClient.getResult(new BearerAccessToken("token"), addFileDocument, None)
+
+    val errorResponse = result.failed.futureValue
+    errorResponse shouldBe a[HttpException]
+
+    val responseException = errorResponse.asInstanceOf[HttpException]
+    responseException.code should equal(StatusCode.Unauthorized)
+    responseException.body should equal("some response body")
+  }
+
+  "a 500 server error response" should "cause an HTTP error" in {
+    wiremockServer.stubFor(post(urlEqualTo(graphQlPath))
+      .willReturn(serverError.withBody("some server error body")))
+
+    val result = getSeriesClient.getResult(new BearerAccessToken("token"), addFileDocument, None)
+
+    val errorResponse = result.failed.futureValue
```

Comment:
I haven't updated the other tests (I'll come back to it if I've got time), but this looks like a nicer way of awaiting a Future value than using `await`. It's non-blocking, and it checks the future periodically up to a timeout that I've defined in `WireMockTest`.